lib/season_statistics.rb

```diff
@@ -1,43 +1,97 @@
 require 'pry'
 
 class SeasonStatistics 
-    def initialize(data)
+  def initialize(data)
         @data = data
-    end
-    def season_game_id_verification 
-        @data[:games].all? do |row|
+   end
+  def season_game_id_verification 
+    @data[:games].all? do |row|
             row[:game_id][0, 4] == row[:season][0, 4]
-        end
-    end
-		def winningest_coach(season)
+  	end
+  end
+	def winningest_coach(season)
 			 (coach_percent(season).max_by{|coach, stat| stat}).shift
-		end
-		def worst_coach(season)
+	end
+	def worst_coach(season)
 			(coach_percent(season).min_by{|coach, stat| stat}).shift
-	 end
-		def coach_percent(season)
-			season_data = @data[:game_teams].select do |row|
-				row[:game_id][0, 4] == season[0, 4]
+	end
+	def most_accurate_team(season) 
+		most_accurate = goal_percentage(season).max_by{|team, average| average}
+			return_team_name(most_accurate[0])
+	end
+	def least_accurate_team(season)
+		least_accurate = goal_percentage(season).min_by{|team, average| average}
+			return_team_name(least_accurate[0])
+	end
+	def most_tackles(season)
+		most_tackles = total_tackles(season).max_by{|team, tackles| tackles}
+			return_team_name(most_tackles[0])
+	end
+	def fewest_tackles(season)
+		fewest_tackles = total_tackles(season).min_by{|team, tackles| tackles}
+		return_team_name(fewest_tackles[0])
+	end
+	def goal_percentage(season)
```

Comment:
hoping we can leverage the goal_percentage, coach_percentage, total_tackles across other methods when we go back and refactor. Methods look good with assistance from the helpers